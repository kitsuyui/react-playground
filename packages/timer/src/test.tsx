import { render } from '@testing-library/react'
import React from 'react'

import { MinimalTimer } from '.'

test('render MinimalTimer', () => {
  const props = {
    remaining: 1000,
    running: false,
    incrementTimerValue: jest.fn(),
    toggle: jest.fn(),
    reset: jest.fn(),
    start: jest.fn(),
    stop: jest.fn(),
    setTimerValue: jest.fn(),
  }
  const { asFragment } = render(<MinimalTimer {...props} />)
  expect(asFragment()).toMatchSnapshot()
})
