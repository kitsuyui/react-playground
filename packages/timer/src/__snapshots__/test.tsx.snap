// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MinimalTimer 1`] = `
<DocumentFragment>
  <div
    style="width: 6em; font-family: monospace;"
  >
    <form>
      <span>
        16:40.000
      </span>
      <br />
      <button
        name="incrementMinute"
        type="submit"
      >
        +分
      </button>
      <button
        name="incrementSecond"
        type="submit"
      >
        +秒
      </button>
      <br />
      <button
        name="reset"
        type="submit"
      >
        Reset
      </button>
      <button
        type="submit"
      >
        Start
      </button>
    </form>
  </div>
</DocumentFragment>
`;
